src/realm_dart_sync.cpp

```diff
@@ -69,20 +66,18 @@ RLM_API void realm_dart_sync_error_handler_callback(realm_userdata_t userdata, r
     } buf;
 
     buf.message = error.error_code.message;
-    error.error_code.message = buf.message.c_str();
-
     buf.detailed_message = error.detailed_message;
-    error.detailed_message = buf.detailed_message.c_str();
-
     buf.user_info_vector.reserve(error.user_info_length);
     for (size_t i = 0; i < error.user_info_length; i++) {
         auto [it, _] = buf.user_info_map.emplace(error.user_info_map[i].key, error.user_info_map[i].value);
         buf.user_info_vector.push_back({ it->first.c_str(), it->second.c_str() });
     }
-    error.user_info_map = buf.user_info_vector.data();
 
     auto ud = reinterpret_cast<realm_dart_userdata_async_t>(userdata);
-    ud->scheduler->invoke([ud, session = *session, error = std::move(error), buf = std::move(buf)]() {
+    ud->scheduler->invoke([ud, session = *session, error = std::move(error), buf = std::move(buf)]() mutable {
```

Comment:
I think the std::move of error is meaningless here as well. In fact all comments carry over.

I am amused that this code is basically identical to the above, but unfortunately, due to the structure of the c api, it is just different enough that I don't think there is a clean way to avoid the code duplication.